### PR TITLE
Remove uses of CPU::Architecture enum in OMRInstOpCodeProperties

### PR DESCRIPTION
**Pull request by dchopra001**

### CODE REVIEW

**Comment by dchopra001** [comment]
@fjeremic FYI I've partitioned the replacements of the enums in different commits so that it would be easier for me to verify the change in the GitHub GUI. I can squash those commits together if you prefer.

**Comment by fjeremic** [comment]
@genie-omr build zlinux,zos

**Comment by fjeremic** [comment]
@dchopra001 looks like there are a bunch of failures related to this change. Could you take a look?

**Comment by dchopra001** [comment]
The PR test failed with an assertion and the backtrace is:
```
29: #5: function OMR::Z::CPU::is_at_least_old_api(OMRProcessorArchitecture)+0x1b2 [0x81998d42]
12:34:46  29: #6: function OMR::Z::CPU::isAtLeast(OMRProcessorArchitecture)+0x66 [0x8199823e]
12:34:46  29: #7: function OMR::Z::Instruction::Instruction(TR::CodeGenerator*, OMR::InstOpCode::Mnemonic, TR::Node*)+0x18a [0x81d60332]
12:34:46  29: #8: function TR::Instruction::Instruction(OMR::InstOpCode::Mnemonic, TR::Node*, TR::CodeGenerator*)+0x4c [0x81c5e194]
```

`is_at_least_old_api` can only be invoked if `TR::Compiler->omrPortLib == NULL`. Most likely the problem happens because `is_at_least_old_api` doesn't know how to handle z9, z990 and z900 type architectures.

**Comment by fjeremic** [comment]
Minimum ALS in OMR is currently z9, so those processor levels are not needed. In fact we are just about to bump it to z10. That could be related to the issue here.

**Comment by dchopra001** [comment]
Right. But some of the instructions defined inside `OMRInstOpCodeProperties.hpp` have their minimumALS set to those processors. If we are to keep that as is, then `is_at_least_old_api` will need to know how to handle those values inside its switch case. 

Or are you suggesting that any instruction whose minimumALS is `z9, z990 and z900` should be bumped up to `z10` as part of this PR?

**Comment by fjeremic** [comment]
> Or are you suggesting that any instruction whose minimumALS is `z9, z990 and z900` should be bumped up to `z10` as part of this PR?

I would do this as a last resort. It's best as you said that we don't "lie" about what processor forst introduced a particular instruction based off of OMR ALS. It's best we do it the right way as you mentioned to fix `is_at_least_old_api`

**Comment by dchopra001** [comment]
Done! Can we please try another run?

**Comment by fjeremic** [comment]
@genie-omr build zlinux,zos

File: compiler/z/env/OMRCPU.cpp [hunk 1]
```diff
@@ -31,6 +31,36 @@
 #include "control/Options_inlines.hpp"
 #include "env/CPU.hpp"
 
+OMR::Z::CPU::CPU(const OMRProcessorDesc& processorDescription) : OMR::CPU(processorDescription)
```
Comment (harryyu1994) [inline compiler/z/env/OMRCPU.cpp:34]: `_supportedArch` is part of the old implementation and should be initialized else where. (In OpenJ9)
Should still use `_supportedArch(z9)` in this constructor.

@dchopra001

Comment (harryyu1994) [reply compiler/z/env/OMRCPU.cpp:34]: Also we don't need to worry about the processor detection assertion test being triggered as currently standalone OMR doesn't use the new implementation (as its omrPortLib pointer is NULL).

Comment (dchopra001) [reply compiler/z/env/OMRCPU.cpp:34]: Then where should `_supportedArch` be initialized? Do we later go and update `_supportedArch`? For context, the issue here happens when the following code is run:

```
#if defined(J9VM_OPT_JITSERVER)
   // In JITServer, we would like to use JITClient's processor info for the compilation
   // The following code effectively replaces the cpu with client's cpu through the getProcessorDescription() that has JITServer support
   if (self()->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
      {
      OMRProcessorDesc JITClientProcessorDesc = TR::Compiler->target.cpu.getProcessorDescription();
      _target.cpu = TR::CPU(JITClientProcessorDesc);
      }
   else
#endif /* defined(J9VM_OPT_JITSERVER) */
```

The constructor that I've modified here is invoked in the above code. And we don't always use this constructor for normal initialization.

Comment (harryyu1994) [reply compiler/z/env/OMRCPU.cpp:34]: `_supportedArch` is initialized in `J9::Z::CPU::initializeS390ProcessorFeatures()`.
Okay I understand why you made this change now.
To make this work, I think we need to add `_target.cpu.initializeS390ProcessorFeatures()` after `_target.cpu = TR::CPU(JITClientProcessorDesc)`.

**Pull request merged**